### PR TITLE
Sensitive type

### PR DESCRIPTION
**Pull request by logicminds**
This PR does two things

1. Keeps the JWT secret from being displayed in pupppet logs, diffs, and puppetdb by using the Sensitive type through the entire compilation and storage cycle.  Users will see `redacted` when the autosign.conf file changes.  The user is not required to pass in a Sensitive type so that we can maintain backwards compatibility when passing in the config parameter.  If they choose to do this the jwt_secret will be exposed in puppetdb.  But the puppet logs will still be wrapped with a Sensitive type.  

Please not this does require puppet 4.6+

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Changelog
```
Comment (danieldreier) [inline CHANGELOG.md]: I think this improvement should probably be in the changelog

File: Rakefile [hunk 2]
```diff
@@ -1,2 +1,88 @@
+# frozen_string_literal: true
```
Comment (danieldreier) [inline Rakefile]: I don't understand how all these Rakefile changes relate to adding Sensitive type

File: .fixtures.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 fixtures:
   repositories:
     stdlib: "https://github.com/puppetlabs/puppetlabs-stdlib.git"
+    debug: 'https://github.com/nwops/puppet-debug.git'
```
Comment (danieldreier) [inline .fixtures.yml]: I don't understand why this is in a change related to adding sensitive type

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1,5 @@
+*.rb eol=lf
```
Comment (danieldreier) [inline .gitattributes]: this seems fine but unrelated to the immediate change at hand

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
```
Comment (danieldreier) [inline .gitlab-ci.yml]: adding gitlab seems entirely unrelated to adding Sensitive type

File: appveyor.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+---
```
Comment (danieldreier) [inline appveyor.yml]: adding appveyor support seems entirely unrelated to adding Sensitive type

File: spec/default_facts.yml [hunk 7]
```diff
@@ -2,7 +2,13 @@
 #
 # Facts specified here will override the values provided by rspec-puppet-facts.
 ---
-concat_basedir: "/tmp"
 ipaddress: "172.16.254.254"
+ipaddress6: "FE80:0000:0000:0000:AAAA:AAAA:AAAA"
```
Comment (danieldreier) [inline spec/default_facts.yml:11]: reasonable improvement but also seems entirely unrelated to adding Sensitive type

**Review by danieldreier — CHANGES_REQUESTED** [review]
@logicminds I really appreciate you adding sensitive type support here. This seems like a very useful and legitimate improvement. I'm having a hard time reviewing for two reasons:
1. There are a bunch of changes that are unrelated to adding Sensitive type, perhaps coming from PDK, and
2. It looks to me like tests are now failing

I don't want to merge this under any circumstances without tests passing. I'd be open to adding the other stuff if there's a good reason; PDK seems legitimate. I don't understand whether there's a good rationale for adding GitLab and Appveyor support. My impulse is that if you're going to add those, we also need to set them up on the repo and maybe add badges, or else those will get stale and give users the misleading impression that there are tested CI configurations available for those CI platforms. I don't know what .sync.yml is for.

My impulse would be to separate these various changes out into different PRs, but I don't want to put a bunch of extra work on you when this is basically fine. If it's easy, let's separate them out and discuss them individually. If not, I'd rather see the work go into fixing tests. I don't have a good puppet dev environment anymore, so I will be 100% dependent on examining the tests to have confidence that this won't break other users.

**Comment by logicminds** [comment]
Guessing you haven't used pdk in a while.  The majority of the changes came from a pdk update. Which if you are not familiar with is like a module sync update.  Since we don't have any customizations such as removing gitlab ci file and appveyor these things will occur.  I will just throw up a separate or for the pdk updates.  They are in unique commits already.

File: .fixtures.yml (hunk 3 above)
Comment (logicminds) [reply .fixtures.yml]: Debug is like pry for puppet   it is only a dev dependency.

**Comment by logicminds** [comment]
@danieldreier updated without pdk changes.

File: templates/autosign.conf.epp [hunk 8]
```diff
@@ -1,5 +1,6 @@
+<%- |
```
Comment (danieldreier) [inline templates/autosign.conf.epp:1]: super super happy to see this as an EPP now!

**Review by danieldreier — APPROVED** [review]
@logicminds this is great work and I really appreciate you contributing it
